packages/taquito-michelson-encoder/src/tokens/comparable/tx_rollup_l2_address.ts

```diff
@@ -0,0 +1,115 @@
+import { BaseTokenSchema } from '../../schema/types';
+import { b58decode, encodePubKey, validateAddress, ValidationResult } from '@taquito/utils';
+import { ComparableToken, SemanticEncoding, Token, TokenFactory, TokenValidationError } from "../token";
+
+export class TxRollupL2AddressValidationError extends TokenValidationError {
+  name = 'TxRollupL2AddressValidationError';
+  constructor(public value: unknown, public token: TxRollupL2AddressToken, message: string) {
+    super(value, token, message)
+  }
+}
+
+export class TxRollupL2AddressToken extends ComparableToken {
+  // should be just l2_address?
+  static prim: 'tx_rollup_l2_address' = 'tx_rollup_l2_address';
+
+  constructor(
+    protected val: { prim: string; args: any[]; annots: any[] },
+    protected idx: number,
+    protected fac: TokenFactory
+  ) {
+    super(val, idx, fac)
+  }
+
+  public ToBigMapKey(val: any) {
+    const decoded = b58decode(val)
```

Comment:
can you add a unit test for ToBigMapKey?
b58decode doesn't support tz4, support needs to be added in the function

Reply:
👍